api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -264,4 +295,498 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() throws Exception {
+		Concept concept = conceptService.getConceptByUuid(drugUuid);
+		List<Drug> drugs = conceptService.getDrugs(drugName, concept, true, true, true, 0, 100);
+		assertTrue(drugs.contains(conceptService.getDrug(drugName)));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrug(String)
+	 * @Verifies should return the matching drug object
+	 */
+	@Test
+	public void getDrug_shouldReturnTheMatchingDrugObject() throws Exception {
+		String drugName = "ASPIRIN";
+		String drugUuid = "05ec820a-d297-44e3-be6e-698531d9dd3f";
+		Drug drug = conceptService.getDrugByUuid(drugUuid);
+		assertEquals(drug, conceptService.getDrug(drugName));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrug(String)
+	 * @Verifies should return null if no matching drug is found
+	 */
+	@Test
+	public void getDrug_shouldReturnNullIfNoMatchingDrugIsFound() throws Exception {
+		int drugIdNotPresent = 1234;
+		assertNull(conceptService.getDrug(drugIdNotPresent));
+	}
+	
+	/**
+	 * @see ConcepTServiceImpl#retireDrug(Drug, String)
+	 * @Verifies should retire the given Drug
+	 */
+	@Test
+	public void retireDrug_shouldRetireTheGivenDrug() throws Exception {
+		String uuidOfDrugToCheck = "05ec820a-d297-44e3-be6e-698531d9dd3f";
+		Drug drug = conceptService.getDrugByUuid(uuidOfDrugToCheck);
+		conceptService.retireDrug(drug, "some dummy reason");
+		assertTrue(drug.isRetired());
+	}
+	
+	/**
+	 * @see ConcepTServiceImpl#unretireDrug(Drug)
+	 * @Verifies should mark Drug as not retired
+	 */
+	@Test
+	public void unretireDrug_shouldMarkDrugAsNotRetired() throws Exception {
+		String uuidOfDrugToCheck = "7e2323fa-0fa0-461f-9b59-6765997d849e";
+		Drug drug = conceptService.getDrugByUuid(uuidOfDrugToCheck);
+		conceptService.unretireDrug(drug);
+		assertFalse(drug.isRetired());
+	}
+	
+	/**
+	 * @see ConcepTServiceImpl#unretireDrug(Drug)
+	 * @Verifies should not change attributes of Drug that is already not retired
+	 */
+	@Test
+	public void unretireDrug_shouldNotChangeAttributesOfDrugThatIsAlreadyNotRetired() throws Exception {
+		String uuidOfDrugToCheck = "3cfcf118-931c-46f7-8ff6-7b876f0d4202";
+		Drug drug = conceptService.getDrugByUuid(uuidOfDrugToCheck);
+		assertFalse(drug.isRetired());
+		conceptService.unretireDrug(drug);
+		assertFalse(drug.isRetired());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getAllConceptClasses()
+	 * @Verifies should return a list of all concept classes
+	 */
+	@Test
+	public void getAllConceptClasses_shouldReturnAListOfAllConceptClasses() throws Exception {
+		int resultSize = 20;
+		List<ConceptClass> conceptClasses = conceptService.getAllConceptClasses();
+		assertEquals(resultSize, conceptClasses.size());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getAllConceptClasses(boolean)
+	 * @Verifies should return a list of all matching concept classes
+	 */
+	@Test
+	public void getAllConceptClasses_shouldReturnAListOfAllMatchingConceptClasses() throws Exception {
+		int resultSizeWhenTrue = 20;
+		int resultSizeWhenFalse = 20;
```

Comment:
@mavrk same as above please split these into separate tests. so you also need two should annotations
for example:
should return a list of all non retired concept classes when given false
should return a list of all concept classes including retired ones when given true